docs/changes/timeseries/18188.perf.rst

```diff
@@ -0,0 +1,2 @@
+Fixed the ``aggregate_downsample`` performance degradation when
```

Comment:
I think we need to rename this `18188.bugfix.rst` to make the changelog entry checker happy. Which I think is actually appropriate - that it effectively is a bug is also why we'll backport it!